src/React/Basic.purs

```diff
@@ -32,25 +29,57 @@ foreign import react_
 -- | constructed using the helper functions provided by the `React.Basic.DOM`
 -- | module (and re-exported here).
 react
```

Comment:
I find this API a little more confusing. If we're renaming things, I like `reactComponent` for creating a `ReactComponent`, but `react` should just go away IMO, and if we want a function for creating JSX directly from a spec (did you have a use case in mind?) then we can probably find a new name for that.

Reply:
That function already exists (`createElement`).  The only difference is whether you use `createElement` at the time of component definition (like the DOM module does) or in a render function.  It's not necessary, but I do think it's more pleasant to use if you're writing lots of PS components.